resources/prosody-plugins/mod_muc_password_check.lua

```diff
@@ -35,145 +35,129 @@ function verify_token(token, room_address)
     session.auth_token = token;
     local verified, reason, msg = token_util:process_and_verify_token(session);
     if not verified then
-        module:log("warn", "not a valid token %s %s", tostring(reason), tostring(msg));
+        module:log("warn", "not a valid token %s %s for %s", tostring(reason), tostring(msg), room_address);
         return false;
     end
 
     return true;
 end
 
-function handle_validate_room_password (event)
-    local request = event.request;
-    module:log("info","Request for room password validate: reqid %s", request.headers["request_id"])
-
-    if request.headers.content_type ~= json_content_type
-            or (not request.body or #request.body == 0) then
-        module:log("warn", "Wrong content type: %s", request.headers.content_type);
-        return 400;
+-- Validates the request by checking for required url param room and
+-- validates the token provided with the request
+-- @param request - The request to validate.
+-- @return [error_code, room]
+local function validate_and_get_room(request)
+    if not request.url.query then
+        module:log("warn", "No query");
+        return 400, nil;
     end
 
-    local params = json.decode(event.request.body);
-    if not params then
-        module:log("warn", "Missing params");
-        return 400;
+    local params = formdecode(request.url.query);
+    local room_name = urlencode(params.room) or "";
+    local subdomain = urlencode(params.prefix) or "";
+
+    if not room_name then
+        module:log("warn", "Missing room param for %s", room_name);
+        return 400, nil;
     end
 
-    local conference = params["conferenceFullName"];
-    local passcode = params["passcode"];
+    local room_address = jid.join(room_name, muc_domain_prefix.."."..muc_domain_base);
 
-    if (not conference or not passcode) then
-        module:log("warn", "Missing conference or passcode param");
-        return 400;
+    if subdomain and subdomain ~= "" then
+        room_address = "["..subdomain.."]"..room_address;
     end
 
-    local room_address = room_jid_match_rewrite(conference)
-
     -- verify access
-    local token = event.request.headers["authorization"]
+    local token = request.headers["authorization"]
 
     if token and starts_with(token,'Bearer ') then
         token = token:sub(8,#token)
     end
 
     if not verify_token(token, room_address) then
-        return 403;
+        return 403, nil;
     end
 
     local room = get_room_from_jid(room_address);
 
-    response_data = {};
-    local PUT_response = {
-        headers = { content_type = "application/json"; };
-    };
-    if room then
-        response_data["valid"] = room:get_password() == passcode;
+    if not room then
+        module:log("warn", "No room found for %s", room_address);
+        return 404, nil;
     else
-        response_data["valid"] = false;
-
-        -- no room found but everything else was valid
-        PUT_response.status_code = 404;
+        return 200, room;
```

Comment:
What is `room` here, just a string? Or would be returning a complex object?

Reply:
It is the room object. An object with methods and such.